README.md

```diff
@@ -220,6 +234,83 @@ charts:
           - linux/arm64
 ```
 
+### Controlling development versions in Chart.yaml
+
+Like some "package version in version control" tools,
+you don't need to manage versions at all in chartpress except by tagging commits.
+
+However, relying only on tags results in "development versions" (versions published from commits after a release)
+having somewhat confusing prerelease versions.
+
+After publishing e.g. `1.2.3`, the next version will be `1.2.3-0.dev.git.1.habc`.
+According to Semantic Versioning,
+this is a "pre release" (good, it should be excluded from default installation),
+but it means it comes _before_ 1.2.3 (wrong! it's 1 commit _newer_ than 1.2.3).
+This is because prereleases should be defined relative to the _next_ release,
+not the _last_ release. But git tags only store the _last_ release!
```

Comment:
This logic only holds for when using stable releases (`x.y.z` release, without pre-release sections), whenever we have `1.2.3-beta.1` for example, appending `.git.1.habcd123` makes it a more recent release.

The release process described below assumes you cut a stable release, but if you cut a pre-release like `1.2.3-beta.1` you shouldn't go back to `1.2.3-beta.2` or `1.2.3-beta.2` etc.

This complexity and UX is what makes me uneasy about accepting this change at this point. I perceive it as very complicated to document and convey the correct usage, to the extent I perceive it will be only you and me that understands how it works and how to use it reliably =/

Reply:
I get that. I want this to be understandable!

I know I'm more comfortable with it, because this is closer to how almost every _other_ project I work with is versioned - make a release, tag it, bump version to next-dev. You _may_ choose to skip the last step if you don't care about folks installing from a dev branch having sensible versions. I don't feel like I'm documenting special chartpress use, I'm documenting widespread standard practice, but the chartpress audience is a bit different.

I'm less concerned, too, because this is all opt-in. It's only important for folks who:

1. publish dev releases (us), and
2. care about dev releases having a sensible version (also us)

I think this is a better experience as both a maintainer and consumer of charts, and I think everyone _should_ opt-in to the new behavior, but the fact that it's optional does add to the complexity. I also want it to be clear that folks are welcome to ignore it - the only cost is improper ordering of prerelease versions (by a computer, humans can understand it fine!).

At a high level, I think the following workflow is not optional for folks who want to have development versions that make sense (whether a helm chart or any other package):

1. set release version
2. publish release (e.g. tbump, chartpress --tag)
3. (if stable release) set next-dev version

We have to consider exactly how to express that, but those three steps must exist somewhere, because there is no way to automate step 3 - the user has to tell _something_ what the next version is going to be. What do you think your ideal process would look like? Maybe we can make it work.